diff --git a/pyproject.toml b/pyproject.toml
index 23168906..eaa523a7 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -36,9 +36,9 @@ keywords = [
 ]
 
 dependencies = [
-    "numpy>=1.22,<2", # Cannot use 2.0 due to matplotlib version pinning.
+    "numpy",
     "scipy",
-    "matplotlib==3.5.1", # Large user-base has apt-installed python3-matplotlib (ROS2) which is pinned to this version.
+    "matplotlib",
     "ansitable",
     "typing_extensions",
     "pre-commit",
